frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java

```diff
@@ -1239,4 +1267,117 @@ private void updateVulnerabilityVerification(ResourceRequest request, ResourceRe
+        for (int i = componentParameters.getDisplayStart(); i < count; i++) {
+            JSONObject jsonObject = JSONFactoryUtil.createJSONObject();
+            Component comp = sortedComponents.get(i);
+            jsonObject.put("id", comp.getId());
+            jsonObject.put("DT_RowId", comp.getId());
+            jsonObject.put("name", SW360Utils.printName(comp));
+            jsonObject.put("cType", nullToEmptyString(comp.getComponentType()));
+            jsonObject.put("lRelsSize", String.valueOf(comp.getReleaseIdsSize()));
+            jsonObject.put("attsSize", String.valueOf(comp.getAttachmentsSize()));
+
+            JSONArray vendorArray = createJSONArray();
+            if (comp.isSetVendorNames()) {
+                comp.getVendorNames().stream().sorted().forEach(vendorArray::put);
+            }
+
+            JSONArray licenseArray = createJSONArray();
+            if (comp.isSetMainLicenseIds()) {
+                comp.getMainLicenseIds().stream().sorted().forEach(licenseArray::put);
+            }
+
+            jsonObject.put("vndrs", vendorArray);
+            jsonObject.put("lics", licenseArray);
+            componentData.put(jsonObject);
+        }
+
+        return componentData;
+    }
+
+    private int getComponentDataCount(PaginationParameters componentParameters, int maxSize) {
+        if (componentParameters.getDisplayLength() == -1) {
+            return maxSize;
+        } else {
+            return min(componentParameters.getDisplayStart() + componentParameters.getDisplayLength(), maxSize);
+        }
+    }
+
+    private List<Component> sortComponentList(List<Component> componentList, PaginationParameters componentParameters) {
+
+        switch (componentParameters.getSortingColumn()) {
+            case COMPONENT_DT_ROW_VENDOR:
+                Collections.sort(componentList, compareByVendor(componentParameters.isAscending()));
+                break;
+            case COMPONENT_DT_ROW_NAME:
+                Collections.sort(componentList, compareByName(componentParameters.isAscending()));
+                break;
+            case COMPONENT_DT_ROW_MAIN_LICENSES:
+                Collections.sort(componentList, compareByMainLicenses(componentParameters.isAscending()));
+                break;
+            case COMPONENT_DT_ROW_TYPE:
+                Collections.sort(componentList, compareByComponentType(componentParameters.isAscending()));
+                break;
+            case COMPONENT_DT_ROW_ACTION:
+                Collections.sort(componentList, compareById(componentParameters.isAscending()));
+                break;
+            default:
+                break;
+        }
+
+        return componentList;
+    }
+
+    private Comparator<Component> compareByVendor(boolean isAscending) {
+        Comparator<Component> comparator = Comparator.comparing(
+                c -> c.isSetVendorNames() ? CommonUtils.COMMA_JOINER.join(c.getVendorNames().stream().sorted().collect(Collectors.toList())) : "");
```

Comment:
minor comment: You could extract that snippet to a function

```java
     private String sortAndConcat(List<String> list) {
         if(list == null || list.length == 0){
             return "";
         }
         return CommonUtils.COMMA_JOINER.join(list.stream().sorted().collect(Collectors.toList()));
     }
```

Reply:
minor comment: I would expect component wise comparison on the list items (after sorting). Although that might not make a big difference.

Reply:
created helper function `private String sortAndConcat(Set<String> strings) `